uefi-test-runner/src/boot/memory.rs

```diff
@@ -17,32 +18,42 @@ pub fn test() {
 }
 
 fn test_allocate_pages() {
-    let num_pages = 1;
-    let ptr =
+    let num_pages = 3;
+    let mut ptr =
         boot::allocate_pages(AllocateType::AnyPages, MemoryType::LOADER_DATA, num_pages).unwrap();
-    let addr = ptr.as_ptr() as usize;
-    assert_eq!(addr % 4096, 0, "Page pointer is not page-aligned");
+
+    let buffer = unsafe { ptr.as_mut() };
```

Comment:
Same issue here, this is unsound with uninitialized memory. I don't think we have to use `write_volatile` like the code here did originally, but we do need to use some pointer function like `ptr::write` or `ptr::write_bytes`.

Since so many of the `NonNull<[T]>`/`*mut [T]` functions are unstable, I think it's most convenient to cast it back to `*mut u8`. Then you can do `unsafe { ptr.write_bytes(...) }`.

Ditto for the docstring examples on `boot::allocate_pages`/`boot::allocate_pool`.

Reply:
Ahh yes, good catch, thanks!